### PR TITLE
add: eam parsers

### PR DESCRIPTION
**Pull request by JulianToledano**

### CODE REVIEW

File: parser/constants.go [hunk 1]
```diff
@@ -90,6 +95,9 @@ const (
 	MethodGetStorageAt                = "GetStorageAt"                // MethodsEVM
 	MethodInvokeContractReadOnly      = "InvokeContractReadOnly"      // MethodsEVM
 	MethodInvokeContractDelegate      = "InvokeContractDelegate"      // MethodsEVM
+	MethodCreate                      = "Create"                      // MethodsEam
+	MethodCreate2                     = "create2"                     // MethodsEam
```
Comment (raynaudoe) [inline parser/constants.go:99]:
```suggestion
	MethodCreate2                     = "Create2"                     // MethodsEam
```

File: parser/eam.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	metadata := make(map[string]interface{})
+	switch txType {
+	case MethodConstructor:
+	case MethodCreate:
+		return p.parseCreate(msg.Params, msgRct.Return, msgCid)
+	case MethodCreate2:
+		return p.parseCreate2(msg.Params, msgRct.Return, msgCid)
+	case MethodCreateExternal:
+		return p.parseCreateExternal(msg, msgRct, msgCid, ethLogs)
+	}
```
Comment (raynaudoe) [inline parser/eam.go:23]: You can use the same parser for the three methods, the return params are the same:
this is from lotus code:
```go
type CreateReturn Return
type Create2Return Return
type CreateExternalReturn Return
```

File: parser/eam.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	metadata[ethLogsKey] = res
+
+	return metadata, nil
+}
+
+// TODO: add cid and short?
+func (p *Parser) appendEamAddress(r eam.Return) {
+	p.appendToAddresses(types.AddressInfo{
+		Robust:    r.RobustAddress.String(),
```
Comment (raynaudoe) [inline parser/eam.go:131]: you can construct the short address like: "f0" + r.ActorId

**Review by raynaudoe — CHANGES_REQUESTED** [review]

File: parser/constants.go (hunk 1 above)
Comment (JulianToledano) [reply parser/constants.go:99]: uhh nice catch!!

**Review by raynaudoe — APPROVED** [review]
Nice!!

**Pull request merged**